### PR TITLE
Documentation updates for 4.2

### PR DESCRIPTION
**Pull request by Craigacp**
### Description
Javadoc and markdown documentation updates for the 4.2 release, along with two new tutorials: onnx export and reproducibility.

### Motivation
New features need docs.

### CODE REVIEW

File: tutorials/external-models-tribuo-v4.ipynb [hunk 1]
```diff
@@ -133,10 +133,7 @@
     }
    ],
    "source": [
-    "var lrTrainer = new LinearSGDTrainer(new LogMulticlass(),new AdaGrad(0.1),3,30000,Trainer.DEFAULT_SEED);\n",
-    "var lrModel = lrTrainer.train(mnistTrain);\n",
-    "var lrEvaluation = labelFactory.getEvaluator().evaluate(lrModel,mnistTest);\n",
-    "lrEvaluation.toString();"
```
Comment (pogren) [inline tutorials/external-models-tribuo-v4.ipynb:139]: this looks like a fat finger.  "b" is not code per se....

**Comment by Mohammed-Ryiad-Eiadeh** [comment]
Congratulations 🎉🎉🎉🎉

File: tutorials/external-models-tribuo-v4.ipynb (hunk 1 above)
Comment (Craigacp) [reply tutorials/external-models-tribuo-v4.ipynb]: Fixed, thanks.

**Review by pogren — APPROVED** [review]
- mostly documentation and other admin changes (e.g. update to .gitignore, new config.xml example files)
- code changes include Model.castModel is now a non-static member method that operates on 'this' rather than a parameter.  Callers of this method updated.
- because there were lots of updates to the jupyter notebooks I fired them all up and rans as much of the code as I could (i.e. without setting up an OCI account.)  Including the reproducibility notebook which requires JDK 17.  Everything checks out!

**Pull request merged**